### DISCUSSION TITLE
how to get the current url uri schema has http or https <server code>

### DISCUSSION THREAD

**Discussion by manishgautammg3994** [serverpod/serverpod#472] [Q&A] [answered]
i thied using
```dart
(  session.server.securityContext == null) ?"something":"else";
```
but i am facing The error
```
default value of an optional parameter must be constant
```
,
also if you can help
```dart
import 'dart:io';

import 'package:serverpod/serverpod.dart';

class ErrorPageWidget extends WidgetRedirect {
 Session session;
 HttpRequest request;
  ErrorPageWidget(this.session, this.request,
      {super.url = (  this.session.server.securityContext == null) ?"http://localhost:8082/error/index.html" : "https://localhost:8082/error/index.html" });
      
}
```
here in the above code i wish to get the real domain url instead of localhost during production and also want to get the security schema , means http or https, 
</br>
## i wish you will find a better solution for this

**Comment by vlidholt** [comment, marked as answer]
Typically, you want to run your Serverpod server as a http server, then handle https through a load balancer. The way the deployment scripts for AWS are setup, that is how things work.
